Magyar olvasdel (#121)
kir-dev/pek-next: README.hu.md

# PéK-Next

**A Schönherz Kollégium és a [SVIE](http://svie.hu/) nyilvántartó rendszere**

[![Build Status](https://travis-ci.org/kir-dev/pek-next.png?branch=master)](https://travis-ci.org/kir-dev/pek-next)
[![Code Climate](https://codeclimate.com/github/kir-dev/pek-next.png)](https://codeclimate.com/github/kir-dev/pek-next)
[![Dependency Status](https://gemnasium.com/kir-dev/pek-next.png)](https://gemnasium.com/kir-dev/pek-next)

## Rendszerkövetelmények

- Ruby 2.5.7
- Postgresql 9.6
- Node (asset fordítás)
- Redis (választható)

vagy

- Docker

> ## Telepítési függőségek

Reviewer: Why don't we use something more natural?
```suggestion
## Függőségek telepítése
```

### Debian alapú rendszerekhez

###### Csomagok

```bash
# A Redis választható
sudo apt install postgresql-9.6 libpq-dev nodejs redis-server
```

###### Ruby 2.5.7

> [Asdf](https://asdf-vm.github.io/asdf/#/core-manage-asdf-vm) használata ajánloatt [ruby pluginnel](https://github.com/asdf-vm/asdf-ruby) vagy [rbenv](https://github.com/rbenv/rbenv) önmagában. Telepítsd a Ruby 2.5.7-et, majd állítsd be a futtatott verziószámot. Az éppen futó verziót a `ruby -v` paranccsal tudod lekérdezni.

Reviewer: ```suggestion
[Asdf](https://asdf-vm.github.io/asdf/#/core-manage-asdf-vm) használata ajánlott [ruby pluginnel](https://github.com/asdf-vm/asdf-ruby) vagy [rbenv](https://github.com/rbenv/rbenv) önmagában. Telepítsd a Ruby 2.5.7-et, majd állítsd be a futtatni kívánt verziót. Az éppen futó verziót a `ruby -v` paranccsal tudod ellenőrizni.
```

### MacOS

###### Brew

> A legegszerűbb, ha telepítjük a [brew-t](https://brew.sh), majd ezzel a további függőségeket.

Reviewer: ```suggestion
A legegyszerűbb, ha telepítjük a [brew-t](https://brew.sh) csomagkezelőt, majd ezzel a további függőségeket.
```

```bash
# A Redis választható
brew install asdf postgresql@9.6 redis
brew services start postgresql@9.6
brew services start redis
```

###### Ruby 2.5.7 és Node

> Add hozzá a [ruby](https://github.com/asdf-vm/asdf-ruby) és [node](https://github.com/asdf-vm/asdf-nodejs) bővítményeket az asdfhez. Telepítsd a Ruby 2.5.7-et, majd állítsd be a futtatott verziószámot. Az éppen futó verziót a `ruby -v` paranccsal tudod lekérdezni.

Reviewer: ```suggestion
Add hozzá a [ruby](https://github.com/asdf-vm/asdf-ruby) és [node](https://github.com/asdf-vm/asdf-nodejs) bővítményeket az asdfhez. Telepítsd a Ruby 2.5.7-et, majd állítsd be a futtatni kiívánt verziót. Az éppen futó verziót a `ruby -v` paranccsal tudod ellenőrizni.
```

## Beállítás

###### Forráskód és függőségek

```bash
git clone https://github.com/kir-dev/pek-next.git
cd pek-next
gem install bundler
bundle install
```

###### Futtatókörnyezet

> Az éles értékek behelyettesítésével hozd létre a `.env` fájlt a `.env.example` alapján.

Reviewer: ```suggestion
Az használni kívánt értékekkel töltsd fel a `.env` fájlt a `.env.example` alapján.
```

###### Adatbázis

```bash
sudo su postgres
psql -c 'create user "pek-next" with superuser password '\''pek-next'\'';'
```

###### Adatbázisséma telepítése

_Saját felhasználóként_

```bash
rake db:setup
```

## Futtatás

###### Szerver

```bash
rails s
```

###### Feldolgozó (választható, redis-t követel)

```bash
bundle exec sidekiq
```

###### Tesztek futtatása

```bash
rake test
```

## Telepítés

A legegyszerűbb mód a docker-compose használata.

> Az éles értékek behelyettesítésével hozd létre a `.env` fájlt a `.env.example` alapján. Az `APP_ID` és `APP_SECRET` változót az auth.sch-ról kapott token alapján vedd fel; a `SECRET_KEY_BASE`-t a `bundle exec rake secret` parancs kimenetével tudod feltölteni.

Reviewer: ```suggestion
A használni kívánt értékekkel töltsd fel a `.env` fájlt a `.env.example` alapján. Az `APP_ID` és `APP_SECRET` változót az auth.sch-ról kapott értékekkel töltsd ki; majd a `SECRET_KEY_BASE`-t a `bundle exec rake secret` parancs segítségével tudod legenerálni.
```
